app/src/main/java/com/nio/CommandProcessorImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.nio;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.SocketChannel;
+import java.nio.charset.StandardCharsets;
+
+//ensure single-responsibility
+public class CommandProcessorImpl implements CommandProcessor {
+
+    public static final String START_READ = "start-read";
+    public static final String STOP_READ = "stop-read";
+
+    public void processCommand(SocketChannel client, String command, CommandHandler handler) {
+        System.out.println("Received command: " + command);
+        if (STOP_READ.equalsIgnoreCase(command.trim())) {
```

Comment:
OCP is violated. Let consider case when we need to add new command in the future or delete existing.